juju_verify/verifiers/ovn_central.py

```diff
@@ -0,0 +1,399 @@
+        """
+        try:
+            status_dict = yaml.safe_load(raw_status)
+            if not isinstance(status_dict, dict):
+                raise ValueError("Input data is not a YAML dictionary.")
+        except (yaml.YAMLError, ValueError) as exc:
+            raise JujuVerifyError(
+                f"Failed to load OVN status as YAML:\n{raw_status}"
+            ) from exc
+
+        try:
+            self.cluster_id = status_dict["Cluster ID"]
+            self.server_id = status_dict["Server ID"]
+            self.status = status_dict["Status"]
+            self.role = status_dict["Role"]
+            self.term = status_dict["Term"]
+            self.leader = status_dict["Leader"]
+            self.vote = status_dict["Vote"]
+            self.log = status_dict["Log"]
+            self.entries_not_committed = status_dict["Entries not yet committed"]
+            self.entries_not_applied = status_dict["Entries not yet applied"]
+            self.servers = status_dict["Servers"]
+
+        except KeyError as exc:
+            raise JujuVerifyError(
+                f"Failed to deserialize OVN cluster status. Missing "
+                f"'{exc.args[0]}' from cluster status data."
+            ) from exc
+
+    @property
+    def short_id(self) -> str:
+        """Return first 4 characters of a Server ID.
+
+        This format is often used to refer to servers within cluster instead of full ID.
+        """
+        return self.server_id[:4]
+
+    @property
+    def is_leader(self) -> bool:
+        """Return true if Server that reported this status consider itself a leader."""
+        return self.leader == "self"
+
+    def __eq__(self, other: Any) -> bool:
+        """Implement equality comparison for ClusterStatus instances."""
+        if not isinstance(other, ClusterStatus):
+            return False
+
+        comparable_attrs = [
+            "cluster_id",
+            "server_id",
+            "status",
+            "role",
+            "term",
+            "leader",
+            "vote",
+            "log",
+            "entries_not_committed",
+            "entries_not_applied",
+            "servers",
+        ]
+        equal = True
+        try:
+            for attr in comparable_attrs:
+                if self.__getattribute__(attr) != other.__getattribute__(attr):
+                    equal = False
+                    break
+        except AttributeError:
+            equal = False
+
+        return equal
+
+
+class UnitClusterStatus(NamedTuple):
+    """Convenience class that groups Southbound and Northbound cluster status.
+
+    This is how a single ovn-central unit views status of these clusters.
+    """
+
+    southbound: ClusterStatus
+    northbound: ClusterStatus
+
+    def __eq__(self, other: Any) -> bool:
+        """Implement equality comparison between UnitClusterStatus instances."""
+        if not isinstance(other, UnitClusterStatus):
+            return False
```

Comment:
We should return [NotImplemented](https://docs.python.org/3.6/library/constants.html#NotImplemented) here.